src/lib/components/ToolsMenu.svelte

```diff
@@ -50,6 +56,10 @@
 					>
 				{/if}
 			</div>
+			<div class="flex space-x-4 text-sm text-gray-500">
+				<button on:click={() => setAllTools(true)}>Enable all</button>
+				<button on:click={() => setAllTools(false)}>Disable all</button>
+			</div>
```

Comment:
```suggestion
			<div class="flex space-x-4 text-sm text-gray-500">
				<button
					disabled={allToolsEnabled}
					class:opacity-50={allToolsEnabled}
					on:click={() => setAllTools(true)}>Enable all</button
				>
				<button
					disabled={allToolsDisabled}
					class:opacity-50={allToolsDisabled}
					on:click={() => setAllTools(false)}>Disable all</button
				>
			</div>
```

and define after `$: activeToolCount` definition
```ts
	$: allToolsEnabled = activeToolCount === $page.data.tools.length;
	$: allToolsDisabled = activeToolCount === 0;
```

for a better UX

Reply:
didn't try but I guess it should use <input type="checkbox"> with indeterminate state
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes

Reply:
![image](https://github.com/huggingface/chat-ui/assets/10467983/bbb174f5-bda0-4804-a516-64ccde4fcfa3)
![image](https://github.com/huggingface/chat-ui/assets/10467983/55674a64-ce26-4fbc-bb08-30d44571c3b8)

thanks Mishig, looks great imo!

Reply:
could be smaller I can tweak it a bit 👀